### PR TITLE
Optimise fn_numberText

### PR DESCRIPTION
**Pull request by Gnashes**

### CODE REVIEW

**Comment by BoGuu** [comment]
Hi,

The main usage of this function is to display scientific notation correctly.

Using toFixed 0 will most likely break that, did you double check?

Thanks for the PR

**Comment by Gnashes** [comment]
Using ``12345678987654321.12345 call {code}``:

Master Branch:

> Result:
0.29656 ms
Cycles:
3372/10000
Return:
"12,345,679,481,405,440"
> 

My merge request:

> Result:
0.0165 ms
Cycles:
10000/10000
Result:
"12,345,679,481,405,440"
> 

> The main usage of this function is to display scientific notation correctly

That's not really true; as this function exists primarily to convert a number into a string and insert commas every 3rd number (when reading from right to left). This function is pretty much exclusively used for displaying price information.

File: Altis_Life.Altis/core/functions/fn_numberText.sqf [hunk 1]
```diff
@@ -1,28 +1,23 @@
...
+    Convert a number into string (avoiding scientific notation) with comma formatting
 
     Parameter(s):
     _this: NUMBER
 
     Returns:
     STRING
 */
-private ["_number","_mod","_digots","_digitsCount","_modBase","_numberText"];
+params [["_number",0,[0]],"_return"];
```
Comment (DomT602) [inline Altis_Life.Altis/core/functions/fn_numberText.sqf]: Unsure of the purpose of _return here?

Comment (Gnashes) [reply Altis_Life.Altis/core/functions/fn_numberText.sqf]: Ahh, forgot to delete that bit. Feel free to remove it.

It was originally just an alternative (lazy) way to define a variable as private.
`params["_return"];
_return = 0;`
is the same as
`private _return = 0;`

**Comment by BoGuu** [comment]
> That's not really true; as this function exists primarily to convert a number into a string and insert commas every 3rd number (when reading from right to left). This function is pretty much exclusively used for displaying price information.

Indeed it does add the delimiter, but the sci. notation is the key, otherwise it's very unreadable

If people use this for decimals (which they do), they're just going to get an integer, that's an issue

**Comment by Gnashes** [comment]
My merge request does not change the functionality of the code, and improves its performance by ~95% (as shown by the test results). You're certainly free to deny the pull request, but I have no idea why you'd want to.

**Comment by BoGuu** [comment]
If somebody was to use a number < 1, the output should be different with toFixed

File: Altis_Life.Altis/core/functions/fn_numberText.sqf [hunk 2]
```diff
@@ -1,28 +1,23 @@
...
     _this: NUMBER
 
     Returns:
     STRING
 */
-private ["_number","_mod","_digots","_digitsCount","_modBase","_numberText"];
+params [["_number",0,[0]]]; 
+ 
+private _numberText = _number toFixed 0; 
+private _num = (count _numberText);
```
Comment (SimZor) [inline Altis_Life.Altis/core/functions/fn_numberText.sqf]: The parenthesis here are redundant.

File: Altis_Life.Altis/core/functions/fn_numberText.sqf [hunk 3]
```diff
@@ -1,28 +1,23 @@
...
-private ["_number","_mod","_digots","_digitsCount","_modBase","_numberText"];
+params [["_number",0,[0]]]; 
+ 
+private _numberText = _number toFixed 0; 
+private _num = (count _numberText); 
 
-_number = [_this,0,0,[0]] call bis_fnc_param;
-_mod = [_this,1,3,[0]] call bis_fnc_param;
+for "_index" from (_num - 3) to 1 step -3 do { 
+     _numberText = (_numberText select [0, _index]) + "," + (_numberText select [_index]);
```
Comment (SimZor) [inline Altis_Life.Altis/core/functions/fn_numberText.sqf]: Could use format here. You may have used the plus operator here instead if it's faster, but I'm not sure whether thats why you used it here.

**Comment by BoGuu** [comment]
This commit will break any non-integers > 1

**Comment by Gnashes** [comment]
Updated to work with intergers less than 1.

Master Branch:
```
Result:
0.315657 ms
Cycles:
3168/10000
Input/Return:
12345678987654321.12345678987654321 //"12,345,679,481,405,440"
```

```
Result:
0.0313 ms
Cycles:
10000/10000
Input/Return:
.12345678987654321 //"0.123457"
```


My pull request:
```
Result:
0.0206 ms
Cycles:
10000/10000
Input/Return:
12345678987654321.12345678987654321 //"12,345,679,481,405,440"
```

```
Result:
0.0048 ms
Cycles:
10000/10000
Input/Return:
.12345678987654321 //"0.123457"
```


It's also worth noting that the Master Branch does not work with negative integers, despite the primary purpose being to maintain scientific notation. My updated pull request also corrects this issue.

Master Branch:
```
Result:
0.034 ms
Cycles:
10000/10000
Input/Return:
-12345678987654321.12345678987654321 //"-1.23457e+016"
```

My Pull Request:
```
Result:
0.0064 ms
Cycles:
10000/10000
Input/Return:
-12345678987654321.12345678987654321 //"-12345679481405440"
```

**Comment by Gnashes** [comment]
Update pushed. The following are the results of my testing.

Master Branch:
```
0.123456789 //.0329ms | "0.123457"
123456.123456 //.0976ms | "123,456"
0.12345678987654321 //.0323ms | "0.123457"
123456789.123456789 //.146133ms | "123,456,792"
123456.12345678987654321 //.100241ms | "123,456"
12345678987654321.123456 //.300752ms | "12,345,679,481,405,440"
12345678987654321.123456789 //.299043ms | "12,345,679,481,405,440"
12345678987654321.12345678987654321 //.29994ms | "12,345,679,481,405,440"
```

My Revision:
```
0.123456789 //.0128ms | "0.123457"
123456.123456 //.0104ms | "123456"
0.12345678987654321 //.0131ms | "0.123457"
123456789.123456789 //.0143 | "123,456,792"
123456.12345678987654321 //.0101ms | "123456"
12345678987654321.123456 //.02ms | "12,345,679,481,405,440"
12345678987654321.123456789 //.0193ms | "12,345,679,481,405,440"
12345678987654321.12345678987654321 //.0197ms | "12,345,679,481,405,440"
```

**Comment by DomT602** [comment]
As the fork has been deleted, this PR will be closed.

**Pull request closed without merging**